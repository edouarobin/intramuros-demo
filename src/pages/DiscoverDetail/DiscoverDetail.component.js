// @flow

import React, { PureComponent } from 'react';
import {
  StyleSheet,
  View,
  Text,
  Image,
  ScrollView,
  StatusBar,
  TouchableOpacity,
} from 'react-native';

import {
  GoBackButtonWithGradient,
  AddressLabel,
  ContactWeb,
  Schedule,
  Price,
  ImageZoomModal,
} from 'Intramuros/src/components';
import globalStyle, {
  eventDetailImageHeight,
  STATUS_BAR_HEIGHT,
  BIG_SCREEN_PADDING_OFFSET,
} from 'Intramuros/src/style/globalStyle';
import { MyFastImage } from 'Intramuros/src/components';
import HTMLText from 'Intramuros/src/utils/HTMLText';

type PropsType = {
  navigation: NavigationPropsType<{
    pointOfInterestDetails: PointOfInterestType,
  }>,
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: globalStyle.colors.lightGrey,
    flexGrow: 1,
  },
  titleStyle: {
    fontFamily: 'Lato-Bold',
    fontSize: 20,
    // fontWeight: 'bold',
    color: 'black',
    lineHeight: 28,
    marginTop: 18,
    textAlign: 'center',
  },
  titleContainer: {
    marginHorizontal: 16,
    alignItems: 'center',
  },
  categoryStyle: {
    fontFamily: 'Lato-Regular',
    marginTop: 5,
    fontSize: 14,
    color: globalStyle.colors.greyishBrown,
    textAlign: 'center',
  },
  image: {
    width: '100%',
    height: eventDetailImageHeight + STATUS_BAR_HEIGHT,
  },
  descriptionContainer: {
    backgroundColor: 'white',
    flex: 1,
    padding: 16,
    paddingHorizontal: 16 + BIG_SCREEN_PADDING_OFFSET,
  },
  decriptionStyle: {
    fontFamily: 'Lato-Regular',
    fontSize: 14,
    lineHeight: 24,
    color: 'black',
  },
  linksStyle: {
    marginTop: 16,
  },
  moreInfo: {
    fontFamily: 'Lato-BoldItalic',
    color: globalStyle.colors.greyishBrown,
    marginTop: 10,
    marginBottom: 5,
  },
});

export default class DiscoverDetail extends PureComponent<PropsType> {
  state = {
    modalVisible: false,
  };

  openZoomImage = () => {
    this.setState({ modalVisible: true });
  };
  closeImageModal = () => {
    this.setState({ modalVisible: false });
  };

  render() {
    const pointOfInterestDetails = this.props.navigation.state.params
      .pointOfInterestDetails;

    return (
      <View style={{ flex: 1, backgroundColor: 'white' }}>
        <StatusBar
          barStyle="dark-content"
          translucent
          backgroundColor="transparent"
          animated={true}
        />
        <GoBackButtonWithGradient
          navigation={this.props.navigation}
          gradientColor="rgba(256,256,256,0.9)"
          goBackColor="black"
        />
        {pointOfInterestDetails ? (
          <ScrollView contentContainerStyle={styles.container}>
            <TouchableOpacity onPress={this.openZoomImage}>
              <MyFastImage
                style={styles.image}
                source={{
                  uri: pointOfInterestDetails.images[0],
                  priority: 'normal',
                }}
                resizeMode={'cover'}
              />
            </TouchableOpacity>
            <View style={styles.titleContainer}>
              <Text style={styles.titleStyle}>
                {pointOfInterestDetails.title}
              </Text>
              <Text style={styles.categoryStyle}>
                {pointOfInterestDetails.category}
                {pointOfInterestDetails.category &&
                pointOfInterestDetails.distanceToSelectedCity
                  ? ' â€” '
                  : null}
                {pointOfInterestDetails.distanceToSelectedCity
                  ? `${pointOfInterestDetails.distanceToSelectedCity} km`
                  : null}
              </Text>
              <AddressLabel
                address={pointOfInterestDetails.addressLabel}
                latitude={pointOfInterestDetails.latitude}
                longitude={pointOfInterestDetails.longitude}
                label={pointOfInterestDetails.title}
              />
            </View>
            <View style={styles.descriptionContainer}>
              <HTMLText
                style={styles.decriptionStyle}
                value={pointOfInterestDetails.description}
              />
              {pointOfInterestDetails.schedule ? (
                <Schedule text={pointOfInterestDetails.schedule} />
              ) : null}
              {pointOfInterestDetails.price ? (
                <Price price={pointOfInterestDetails.price} />
              ) : null}
              {pointOfInterestDetails.url1 ||
              pointOfInterestDetails.url2 ||
              pointOfInterestDetails.url3 ? (
                <View style={styles.linksStyle}>
                  <Text style={styles.moreInfo}>Plus d'informations :</Text>
                  {pointOfInterestDetails.url1 ? (
                    <ContactWeb
                      website={pointOfInterestDetails.url1}
                      fisrtElement={true}
                    />
                  ) : null}
                  {pointOfInterestDetails.url2 ? (
                    <ContactWeb website={pointOfInterestDetails.url2} />
                  ) : null}
                  {pointOfInterestDetails.url3 ? (
                    <ContactWeb website={pointOfInterestDetails.url3} />
                  ) : null}
                </View>
              ) : null}
            </View>
          </ScrollView>
        ) : null}
        {pointOfInterestDetails ? (
          <ImageZoomModal
            image={pointOfInterestDetails.images[0]}
            modalVisible={this.state.modalVisible}
            closeImageModal={this.closeImageModal}
          />
        ) : null}
      </View>
    );
  }
}
