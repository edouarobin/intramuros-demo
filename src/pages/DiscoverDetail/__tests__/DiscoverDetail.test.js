// @flow-weak

import React from 'react';
import renderer from 'react-test-renderer';

import DiscoverDetail from '../DiscoverDetail.component';

it('Fully renders the DiscoverDetails page', () => {
  const mockPointOfInterestDetails = {
    title: 'Château de Brézé',
    description: 'qzf hzmEF HQZMEF QMZOEF ZQMF ',
    category: 'Château',
    createdAt: '2017-11-16T15:42:46.804527Z',
    updatedAt: '2017-11-16T22:51:24.862558Z',
    city: 1,
    images: ['https://el…', 'https://el…'],
    addressLabel: '2 Rue du Château, 49260 Brézé, France',
    latitude: 47.1746017,
    longitude: -0.0581200999999965,
    id: 4,
    distanceToSelectedCity: 10,
  };
  const tree = renderer
    .create(
      <DiscoverDetail
        navigation={{
          state: {
            params: { pointOfInterestDetails: mockPointOfInterestDetails },
          },
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
