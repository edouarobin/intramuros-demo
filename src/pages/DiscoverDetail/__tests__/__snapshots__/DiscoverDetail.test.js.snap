// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fully renders the DiscoverDetails page 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <BVLinearGradient
    colors={
      Array [
        3875536895,
        16777215,
      ]
    }
    end={undefined}
    locations={null}
    start={undefined}
    style={
      Object {
        "flex": 1,
        "height": 320.16,
        "position": "absolute",
        "width": "100%",
        "zIndex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "paddingHorizontal": 8,
          "paddingTop": 30,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": 100,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "width": 100,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 36,
                },
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "black",
                    "marginHorizontal": 8,
                  },
                  Object {
                    "color": "black",
                  },
                ],
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "black",
                  "fontSize": 16,
                },
                Object {
                  "color": "black",
                },
              ]
            }
          >
            Retour
          </Text>
        </View>
      </View>
    </View>
  </BVLinearGradient>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "backgroundColor": "#F8F8F8",
        "flexGrow": 1,
      }
    }
  >
    <View>
      <Image
        source={
          Object {
            "uri": "https://el…",
          }
        }
        style={
          Object {
            "height": 555.6,
            "width": "100%",
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "marginHorizontal": 16,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "black",
              "fontSize": 20,
              "fontWeight": "bold",
              "lineHeight": 28,
              "marginTop": 18,
            }
          }
        >
          Château de Brézé
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#535353",
              "fontSize": 14,
              "marginTop": 5,
              "textAlign": "center",
            }
          }
        >
          Château
           — 10 km
        </Text>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "flexDirection": "row",
              "marginBottom": 28,
              "marginTop": 17,
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Image
            source="test-file-stub"
            style={
              Object {
                "height": 20,
                "width": 14,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#2869e5",
                "marginLeft": 10,
              }
            }
          >
            2 Rue du Château, 49260 Brézé
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "flex": 1,
            "padding": 16,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "black",
              "fontSize": 14,
              "lineHeight": 24,
            }
          }
        >
          qzf hzmEF HQZMEF QMZOEF ZQMF 
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
